pipelines/cellprofiler/cellprofiler_pipeline.wdl

```diff
@@ -36,86 +36,98 @@ workflow cellprofiler_pipeline {
   String input_directory = sub(input_directory_gsurl, "/+$", "")
   String output_directory = sub(output_directory_gsurl, "/+$", "")
 
-  # Define the input files, so that we use Cromwell's automatic file localization
-  call util.gsutil_ls_to_file as directory {
+  # check write permission on output bucket
+  call util.gcloud_is_bucket_writable as permission_check {
     input:
-      directory_gsurl=input_directory,
-      file_extension=file_extension,
+      gsurls=[output_directory],
   }
 
-  # The single VM workflow
-  if (!do_scatter) {
+  # run the compute only if output bucket is writable
+  Boolean is_bucket_writable = permission_check.is_bucket_writable
+  if (is_bucket_writable) {
```

Comment:
Since this pipeline allows optional delocalization of inputs, the bucket write permission check should only occur if the parameter had a value that was not the empty string.

(btw, [here's another way](https://github.com/openwdl/wdl/blob/main/versions/development/SPEC.md#optional-types-and-none) to define and evaluate optional parameters)

Reply:
Thank you!   Good point